backend/shork/src/main/kotlin/internal/instruction/Sne.kt

```diff
@@ -12,10 +12,45 @@ internal class Sne(
     modifier: Modifier,
 ) : AbstractInstruction(aField, bField, addressModeA, addressModeB, modifier) {
     override fun execute(process: AbstractProcess) {
-        TODO("Not yet implemented")
+        val core = process.program.shork.memoryCore
+        val sourceAddress = resolve(process, aField, addressModeA)
+        val destinationAddress = resolve(process, bField, addressModeB)
+        val sourceInstruction = core.loadAbsolute(sourceAddress)
+        val destinationInstruction = core.loadAbsolute(destinationAddress)
+
+        val match =
+            when (modifier) {
+                Modifier.A -> {
+                    sourceInstruction.aField != destinationInstruction.aField
+                }
+                Modifier.B -> {
+                    sourceInstruction.bField != destinationInstruction.bField
+                }
+                Modifier.AB -> {
+                    sourceInstruction.aField != destinationInstruction.bField
+                }
+                Modifier.BA -> {
+                    sourceInstruction.bField != destinationInstruction.aField
+                }
+                Modifier.F -> {
+                    sourceInstruction.aField != destinationInstruction.aField &&
+                        sourceInstruction.bField != destinationInstruction.bField
+                }
+                Modifier.X -> {
+                    sourceInstruction.aField != destinationInstruction.bField &&
+                        sourceInstruction.bField != destinationInstruction.aField
+                }
```

Comment:
These should use an OR operator instead of an AND. The skip happens if any pair of them aren't equal. I forgot to add this to the wiki though, this will be added shortly.